Dockerfile

```diff
@@ -1,49 +1,47 @@
-
-RUN     wget https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz && \
-          tar -zxf node-v$NODE_VERSION-linux-x64.tar.gz -C /usr/local && \
-          ln -sf /usr/local/node-v$NODE_VERSION-linux-x64 /usr/local/node && \
-          ln -sf /usr/local/node/bin/npm /usr/local/bin/ && \
-          ln -sf /usr/local/node/bin/node /usr/local/bin/ && \
-          rm node-v$NODE_VERSION-linux-x64.tar.gz
-
-# npmrc is created by env-config.sh. For local testing, an empty one is generated
-RUN     touch /usr/local/etc/npmrc && \
-          mkdir /usr/local/node/etc && \
-          cp /usr/local/etc/npmrc /usr/local/node/etc/npmrc
-
-ENV     NODE_PATH /calypso/server:/calypso/client
-
-# Install base npm packages to take advantage of the docker cache
-COPY    ./package.json /calypso/package.json
-COPY    ./npm-shrinkwrap.json /calypso/npm-shrinkwrap.json
-# Sometimes "npm install" fails the first time when the cache is empty, so we retry once if it failed
-RUN     npm install --production || npm install --production
-
-COPY     . /calypso
-
-# Build javascript bundles for each environment and change ownership
-RUN     CALYPSO_ENV=wpcalypso make build-wpcalypso && \
-          CALYPSO_ENV=horizon make build-horizon && \
-          CALYPSO_ENV=stage make build-stage && \
-          CALYPSO_ENV=production make build-production && \
-          chown -R nobody /calypso
-
-USER    nobody
-CMD     NODE_ENV=production node build/bundle-$CALYPSO_ENV.js
+WORKDIR    /calypso
+
+# includes three key files plus everything else
+#   - env-config.sh
+#      used by systems to overwrite some defaults
+#      such as the apt and npm mirrors
+#
+#   - package.json
+#      provides the build scripts needed by npm
+#
+#   - npm-shrinkwrap.json
+#      provides the actual node dependencies for the project
+COPY       . /calypso
+
+ENV        NODE_PATH=/calypso/server:/calypso/client
+
+RUN        true \
+           && mv ./env-config.sh /tmp/env-config.sh \
+           && apt-get -y update \
+           && apt-get -y install \
+                 git \
+                 make \
+           # Sometimes "npm install" fails the first time when the
+           # cache is empty, so we retry once if it failed
+           && npm install --production || npm install --production \
+           && CALYPSO_ENV=wpcalypso make build-wpcalypso \
+           && CALYPSO_ENV=horizon make build-horizon \
+           && CALYPSO_ENV=stage make build-stage \
+           && CALYPSO_ENV=production make build-production \
+           && chown -R nobody /calypso \
+           # Clean up the dependencies we no longer need
+           && apt-get remove -y \
+                 git \
+           && apt-get autoremove -y \
+           && rm -rf /var/lib/apt/lists/* \
+           && rm -rf /var/lib/dpkg/info/* \
+           && rm -rf /tmp/* \
+           && rm -rf /var/tmp/* \
+           && rm -rf /usr/share/{doc,doc-base,info,locale,man}/* \
+           && rm -rf /root/.npm \
```

Comment:
nope. those are created from inside the Docker image during things like `npm install`

I have pulled out the things we copy in via `.dockerignore` though